tremor-script/src/ast.rs

```diff
@@ -489,26 +492,41 @@ pub type Warnings = std::collections::BTreeSet<Warning>;
 
 /// don't use
 #[allow(clippy::struct_excessive_bools)]
+#[derive(Clone)]
 pub struct Helper<'script, 'registry>
 where
     'script: 'registry,
 {
+    // NOTE We should refactor this into multiple structs as its becoming a bit of a
```

Comment:
:100: Very good call, perhaps something nested? `Helper{definitions:Definitions, docs...}`

Reply:
As i sketched out the basic implementation of troy its the helper where quite a lot of the tension and resistance to refactoring came from. The other locus of tension/resistance is module scoping and referencing. I suspect we can go a long way to cleaning this up through a mini refactoring